docs-search/src/Docs/Search/PackageIndex.purs

```diff
@@ -56,14 +59,11 @@ mkPackageInfo packageScores pms =
       { name: packageName
       , description: description
       , score: getPackageScoreForPackageName packageScores packageName
-      , dependencies:
-          unwrap dependencies <#>
-            _.packageName >>> RawPackageName >>> normalizePackageName
-      , repository: repository <#> (_.url)
+      , dependencies: coerce $ dependencies <#> Tuple.fst
+      , repository: repository <#> unwrap >>> (_.url)
       }
-
     where
-    packageName = normalizePackageName $ RawPackageName name
+    packageName = coerce name
```

Comment:
The left hand side is a `PackageName` from `Docs.Search.Types`. The right hand side `name` is a `PackageName` from `Bower.Web.PackageMeta`.
Both are functionally the same, newtypes around string.

You might ask, why am I not using the latter? 

Well it is because the latter type does not implement `EncodeJson` or `DecodeJson`. 

Much of this library relies on generic deriving of these classes. For example, `PackageInfo` contains `PackageName` if I change out the definitions, then the generic instances for `PackageInfo` break.

Reply:
Ah yes, great points. We should:
- not rely on generic Json derivation, because it's opaque - if the type itself changes then it's hard to see how the Json conversion changes as well, which means having explicit implementations for Json conversions
- use [codecs](https://pursuit.purescript.org/packages/purescript-codec-argonaut/10.0.0) instead of typeclasses for encoding json - this is how it's done in the rest of the Spago codebase, you can see examples if you search for `JsonCodec`

This is a small patch so I wouldn't mind doing the porting here, but we can also take it in a new PR if you'd like

Reply:
There are 13 types in `docs-search` that derive generic Encode/Decode Json instances. 
Instead of changing just 1 or 2 in this PR, can I do separate PR to convert all those instances?

Reply:
Of course!